bugz/client.go

```
@@ -23,45 +23,241 @@ type BugzLoginResponse struct {
 	Token string `json:"token"`
 }
 
-func NewBugzClient() BugzClient {
-	login, ok := os.LookupEnv("BUGZILLA_LOGIN")
-	if !ok {
-		panic("BUGZILLA_LOGIN is not set")
-	}
-	password, ok := os.LookupEnv("BUGZILLA_PASSWORD")
-	if !ok {
-		panic("BUGZILLA_PASSWORD is not set")
+func NewBugzClient() *BugzClient {
+	login := os.Getenv("BUGZILLA_LOGIN") //Retrieve env var values and check if they are empty
+	password := os.Getenv("BUGZILLA_PASSWORD")
+	if login == "" || password == "" {
+		panic("BUGZILLA_LOGIN or BUGZILLA_PASSWORD is not set")
 	}
-	var loginResponse BugzLoginResponse
 
-	bugz := BugzClient{
-		url:   "https://bugs.freebsd.org/bugzilla/rest",
+	bc := &BugzClient{
+		URL:   "https://bugs.freebsd.org/bugzilla/rest",
 		token: "",
-		http:  &http.Client{
-			// Timeout: time.Second * 10,
-		},
+		http:  &http.Client{},
 	}
+
 	formData := url.Values{}
 	formData.Set("login", login)
 	formData.Set("password", password)
 
-	response, err := bugz.http.Get(bugz.url + "/login?" + formData.Encode())
-	CheckFatal("NewBugzClient bugz.http.Get", err)
+	response, _ := bc.http.Get(bc.URL + "/login?" + formData.Encode())
```

do not ignore the error here